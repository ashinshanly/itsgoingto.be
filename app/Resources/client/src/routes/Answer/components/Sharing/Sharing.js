import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { pollSelector } from '../../../../store/poll'
import './Sharing.scss'

export const Sharing = ({ poll }) => {
  // Select the text contents from target of the given event
  const selectText = (event) => {
    var doc = document,
      text = event.target,
      range,
      selection
    if (doc.body.createTextRange) {
      range = document.body.createTextRange()
      range.moveToElementText(text)
      range.select()
    } else if (window.getSelection) {
      selection = window.getSelection()
      range = document.createRange()
      range.selectNodeContents(text)
      selection.removeAllRanges()
      selection.addRange(range)
    }
  }

  // Generate link for sharing to twitter
  const twitterLink = () =>
    'https://twitter.com/home?status=' + poll.question + ' - Answer this poll at ' + window.location.href

  // Generate link for sharing to facebook
  const facebookLink = () =>
    'https://www.facebook.com/sharer/sharer.php?u=' + window.location.href

  // Generate link for sharing via email
  const emailLink = () =>
    'mailto:?&subject=' + poll.question + '&body=Answer%20this%20poll%20at%20' + window.location.href

  return (
    <div>
      <h3>
        <span
          className='share-link'
          onClick={ selectText }>
          { window.location.href.replace(/(^\w+:|^)\/\//, '') }
        </span>
        <a
          className='shareButton twitter'
          target='_blank'
          href={ twitterLink() }>
          <i className="fa fa-twitter" />
        </a>
        <a
          className='shareButton facebook'
          target='_blank'
          href={ facebookLink() }>
          <i className='fa fa-facebook' />
        </a>
        <a
          className='shareButton email'
          target='_blank'
          href={ emailLink() }>
          <i className='fa fa-envelope' />
        </a>
      </h3>
    </div>
  )
}

Sharing.propTypes = {
  poll : PropTypes.object.isRequired,
}

const mapStateToProps = (state, props) => ({
  poll : pollSelector(state, props.identifier),
})

export default connect(mapStateToProps)(Sharing)
