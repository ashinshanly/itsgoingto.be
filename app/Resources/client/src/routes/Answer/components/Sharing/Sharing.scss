@import "bourbon";
@import "neat";
@import "variables";

h3 {
  position: relative;
}

.share-link {
  margin: 0px;
  font-weight: 300;
  font-size: 1.0rem;
  margin-top: 15px;
  border-radius: 30px;
  box-shadow: inset 0 1px 3px rgba(0, 0, 0, 0.06), 0 0 5px #D4D4D4;
  background: #EAEAEA;
  padding: 0.25em 0.75em;
  display: inline-block;
}

.shareButton {
  display: inline-block;
  width:1.75em;
  padding: 0.25em 0.25em;
  border-radius: 30px;
  font-size: 1rem;
  margin-left: 5px;

  background: #EAEAEA;
  box-shadow: inset 0 1px 3px rgba(0, 0, 0, 0.06), 0 0 5px #D4D4D4;
  color: $base-font-color;

  @media(max-width: $screen-md-min) {
    position: absolute;
    top:55px;
    left:50%;
    margin-left:22.5px;
  }

  &.twitter {
    background:#55acee;
    color: $white;
    @media(max-width: $screen-md-min) {
      margin-left:-57.5px;
    }
  }

  &.facebook {
    background:#3B5998;
    color: $white;
    @media(max-width: $screen-md-min) {
      margin-left:-17.5px;
    }
  }
}
