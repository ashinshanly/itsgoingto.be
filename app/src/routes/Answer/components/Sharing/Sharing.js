import React from 'react'
import PropTypes from 'prop-types'
import './Sharing.scss'

export function Sharing ({ poll }) {
  // Select the text contents from target of the given event
  const selectText = (event) => {
    const doc = document
    const text = event.target
    let range
    let selection

    if (doc.body.createTextRange) {
      range = document.body.createTextRange()
      range.moveToElementText(text)
      range.select()
    } else if (window.getSelection) {
      selection = window.getSelection()
      range = document.createRange()
      range.selectNodeContents(text)
      selection.removeAllRanges()
      selection.addRange(range)
    }
  }

  // Generate link for sharing to twitter
  const twitterLink = () =>
    'https://twitter.com/home?status=' + poll.question + ' - Answer this poll at ' + window.location.href

  // Generate link for sharing to facebook
  const facebookLink = () =>
    'https://www.facebook.com/sharer/sharer.php?u=' + window.location.href

  // Generate link for sharing via email
  const emailLink = () =>
    'mailto:?&subject=' + poll.question + '&body=Answer%20this%20poll%20at%20' + window.location.href

  return (
    <div>
      <h3>
        <span
          className='share-link'
          onClick={selectText}>
          { window.location.href.replace(/(^\w+:|^)\/\//, '') }
        </span>
        <a
          className='shareButton twitter'
          target='_blank'
          href={twitterLink()}>
          <i className='fa fa-twitter' />
        </a>
        <a
          className='shareButton facebook'
          target='_blank'
          href={facebookLink()}>
          <i className='fa fa-facebook' />
        </a>
        <a
          className='shareButton email'
          target='_blank'
          href={emailLink()}>
          <i className='fa fa-envelope' />
        </a>
      </h3>
    </div>
  )
}

Sharing.propTypes = {
  poll : PropTypes.object.isRequired,
}

export default Sharing
