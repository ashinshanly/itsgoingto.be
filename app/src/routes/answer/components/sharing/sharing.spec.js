/* global expect */
import React from 'react'
import { shallow } from 'enzyme'
import Sharing from './sharing'

const props = {
  poll: {
    question: 'This is a question',
  },
}
let wrapper

describe('(Route) answer', () => {
  describe('(Component) sharing', () => {
    beforeEach(() => {
      wrapper = shallow(<Sharing {...props} />)
    })

    describe('(Render)', () => {
      it('matches snapshot', () => {
        expect(wrapper).toMatchSnapshot()
      })
    })
  })
})