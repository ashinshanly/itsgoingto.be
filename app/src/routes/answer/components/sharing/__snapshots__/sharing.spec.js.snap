// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Route) answer (Component) sharing (Render) matches snapshot 1`] = `
<div>
  <h3>
    <span
      className="share-link"
      onClick={[Function]}
    >
      localhost/
    </span>
    <a
      className="shareButton twitter"
      href="https://twitter.com/home?status=This is a question - Answer this poll at http://localhost/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <i
        className="fa fa-twitter"
      />
    </a>
    <a
      className="shareButton facebook"
      href="https://www.facebook.com/sharer/sharer.php?u=http://localhost/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <i
        className="fa fa-facebook"
      />
    </a>
    <a
      className="shareButton email"
      href="mailto:?&subject=This is a question&body=Answer%20this%20poll%20at%20http://localhost/"
      rel="noopener noreferrer"
      target="_blank"
    >
      <i
        className="fa fa-envelope"
      />
    </a>
  </h3>
</div>
`;
